web/src/popover_menus.js

```diff
@@ -272,6 +288,140 @@ export function initialize() {
         },
     });
 
+    tippy_no_propagation("#stream_filters .topic-sidebar-menu-icon", {
+        ...left_sidebar_tippy_options,
+        onShow(instance) {
+            popover_instances.topics_menu = instance;
+            on_show_prep(instance);
+            const elt = $(instance.reference).closest(".topic-sidebar-menu-icon").expectOne()[0];
+            const $stream_li = $(elt).closest(".narrow-filter").expectOne();
+            const stream_id = stream_popover.elem_to_stream_id($stream_li);
+            const topic_name = $(elt).closest("li").expectOne().attr("data-topic-name");
+            const sub = sub_store.get(stream_id);
+            const topic_muted = user_topics.is_topic_muted(sub.stream_id, topic_name);
+            const has_starred_messages =
+                starred_messages.get_count_in_topic(sub.stream_id, topic_name) > 0;
+            const can_move_topic = settings_data.user_can_move_messages_between_streams();
+            const url = $(elt).closest("li").find(".topic-name").expectOne().prop("href");
+            instance.setContent(
+                parse_html(
+                    render_topic_sidebar_actions({
+                        stream_name: sub.name,
+                        stream_id: sub.stream_id,
+                        topic_name,
+                        topic_muted,
+                        can_move_topic,
+                        is_realm_admin: page_params.is_admin,
+                        topic_is_resolved: resolved_topic.is_resolved(topic_name),
+                        color: sub.color,
+                        has_starred_messages,
+                        url,
+                    }),
+                ),
+            );
+        },
+        onMount(instance) {
+            const $popper = $(instance.popper);
+            $popper.one("click", ".sidebar-popover-mute-topic", (e) => {
+                const stream_id = topic_popover_stream_id(e);
```

Comment:
I've noticed a recurring pattern in our handlers where we require the `stream_id`. I suggest defining it at the `onMount` level of our code and using the same variable throughout. This may require some adjustments to the `topic_popover_stream_id` function, but they are not big. What do you think?

Reply:
Sure, I will refactor it to have only one global variable inside `onMount` for all the use cases.